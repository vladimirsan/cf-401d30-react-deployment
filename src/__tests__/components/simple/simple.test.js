import React from 'react';
import Simple from '../../../components/simple/simple';
import rendeder from 'react-test-renderer'; // Vinicio - snapshot testing


describe("<Simple/>", () => {
  test('basic rendering', () => {
    const mountedSimple = shallow(<Simple/>);
    expect(mountedSimple.find('span')).toBeTruthy();
  });

  test('testing state changes', () => {
    const mountedSimple = mount(<Simple/>);
    const button = mountedSimple.find('button');

    button.simulate('click');
    expect(mountedSimple.state('value')).toBe(true);

    button.simulate('click');
    expect(mountedSimple.state('value')).toBe(false);

    expect(mountedSimple.find('span').text()).toContain('false');
  });

  test('rendering follows the snapshot', () => {
    const snapshot = rendeder.create(<Simple/>).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
