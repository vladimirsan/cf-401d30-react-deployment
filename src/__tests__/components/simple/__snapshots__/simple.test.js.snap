// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Simple/> rendering follows the snapshot 1`] = `
<section
  className="Simple"
>
  <span>
    Value: 
    false
  </span>
  <button
    onClick={[Function]}
  >
    Change Value
  </button>
</section>
`;
